src/monster.cpp

```diff
@@ -659,10 +659,19 @@ void Monster::setIdle(bool idle)
 void Monster::updateIdleStatus()
 {
 	bool idle = false;
-
-	if (conditions.empty()) {
-		if (!isSummon() && targetList.empty()) {
+	if (!isSummon()) {
+		if (conditions.empty() && targetList.empty()) {
 			idle = true;
+		} else if (!conditions.empty()) {
```

Comment:
``if (targetList.empty())`` sounds more reasonable.

Reply:
@slavidodo  I assume you meant else instead of that check at all, why would I check if targlist is empty if then monster could have no conditions at all.

Reply:
but I wouldnt change that at all, but probably first check should be or instead of and

Reply:
If targetList is empty and condition list is empty the first case should have been applied, that means you are probably iterating through conditions when targetList is actually not empty and then idle status should never be modified.

Reply:
@Znote In the first case it checks if both lists are empty (targets and conditions).
the second case would only apply if targets **OR** conditions is not empty, so what he did is checking if conditions is not empty, but, both cases might actually meet. It's logical to check if targets is empty to make sure that we don't iterate over condition list when there are targets that will always stop the idle status.

Reply:
@nekiro My hunch is that is was intention as per: https://github.com/otland/forgottenserver/issues/2410#issuecomment-357823797

Reply:
that's exactly what I mean, you cant set the monster to be idle when target list is not actually empty!

Reply:
please suggest the whole code snippet on how it should be written, my small brain cant handle what you say somehow.

Reply:
Oh I think I got it, because second check always sets idle status. I think doing like 

> if (hasAggressiveCondition) { idle = false; } 
would be fine right?